app/knight.py

```diff
@@ -0,0 +1,65 @@
+from app.gear.armor_part import ArmorPart
+from app.gear.weapon import Weapon
+from app.items.potion import Potion
+
+
+class Knight:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armor: list[ArmorPart],
+            weapon: Weapon,
+            potion: Potion
+    ) -> None:
+        self.name = name
+        self.base_power = power
+        self.power = power
+        self.base_hp = hp
+        self.hp = hp
+        self.armor = armor
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = 0
+        self.apply_gear_and_items()
+
+    @staticmethod
+    def parse_from_dict(knight_dict: dict) -> "Knight":
+        name = knight_dict["name"]
+        power = knight_dict["power"]
+        hp = knight_dict["hp"]
+        armor = [
+            ArmorPart(armor_part["part"], armor_part["protection"])
+            for armor_part in knight_dict["armour"]
+        ]
+        weapon = Weapon(
+            knight_dict["weapon"]["name"],
+            knight_dict["weapon"]["power"]
+        )
+        potion = (
+            Potion(
+                knight_dict["potion"]["name"],
+                knight_dict["potion"]["effect"]
+            )
+            if knight_dict["potion"]
+            else None
+        )
+        return Knight(name, power, hp, armor, weapon, potion)
```

Comment:
basically classmethod is more appropriate here as you return class instance of the same class